From 306abd812cbb86a4bffe3d1aee14eb77cc6b1458 Mon Sep 17 00:00:00 2001
From: Darren Etheridge <detheridge@ti.com>
Date: Fri, 28 Aug 2015 06:18:02 +0000
Subject: [PATCH 8/8] ARM: OMAP2+: Use pdata-quirks for sgx deassert_hardreset

Use pdata_quirks to provide platform data to the sgx driver.  This is used
to provide a function pointer for the sgx driver to access
omap_device_deassert_hardreset along with the reset name as defined in the
corresponding hwmod entry.

This platform data will not be required when a seperate reset driver is
available allowing decoupling from omap_hwmod and omap_device.

Signed-off-by: Darren Etheridge <detheridge@ti.com>
Signed-off-by: Eric Ruei <e-ruei1@ti.com>
---
 arch/arm/mach-omap2/pdata-quirks.c     | 16 +++++++++++++++-
 include/linux/platform_data/sgx-omap.h | 22 ++++++++++++++++++++++
 2 files changed, 37 insertions(+), 1 deletion(-)
 mode change 100644 => 100755 arch/arm/mach-omap2/pdata-quirks.c
 create mode 100644 include/linux/platform_data/sgx-omap.h

diff --git a/arch/arm/mach-omap2/pdata-quirks.c b/arch/arm/mach-omap2/pdata-quirks.c
old mode 100644
new mode 100755
index df2aa81..68b4fff
--- a/arch/arm/mach-omap2/pdata-quirks.c
+++ b/arch/arm/mach-omap2/pdata-quirks.c
@@ -19,6 +19,7 @@
 #include <linux/platform_data/iommu-omap.h>
 #include <linux/platform_data/remoteproc-omap.h>
 #include <linux/platform_data/wkup_m3.h>
+#include <linux/platform_data/sgx-omap.h>
 #include <linux/platform_data/pci-dra7xx.h>
 #include <linux/platform_data/remoteproc-pruss.h>
 
@@ -38,7 +39,12 @@ struct pdata_init {
 
 struct of_dev_auxdata omap_auxdata_lookup[];
 static struct twl4030_gpio_platform_data twl_gpio_auxdata;
-
+#if defined(CONFIG_SOC_AM33XX) || defined(CONFIG_SOC_AM43XX)
+static struct gfx_sgx_platform_data sgx_pdata = {
+	.reset_name = "gfx",
+	.deassert_reset = omap_device_deassert_hardreset,
+};
+#endif
 #if IS_ENABLED(CONFIG_OMAP_IOMMU)
 int omap_iommu_set_pwrdm_constraint(struct platform_device *pdev, bool request,
 				    u8 *pwrst);
@@ -386,6 +392,14 @@ struct of_dev_auxdata omap_auxdata_lookup[] __initdata = {
 	OF_DEV_AUXDATA("ti,am3352-pruss", 0x4a300000, "4a300000.pruss",
 		       &pruss_pdata),
 #endif
+#if defined(CONFIG_SOC_AM33XX)
+	OF_DEV_AUXDATA("ti,am335x-sgx530", 0x56000000, "56000000.sgx",
+		       &sgx_pdata),
+#endif
+#if defined(CONFIG_SOC_AM43XX)
+	OF_DEV_AUXDATA("ti,am437x-sgx530", 0x56000000, "56000000.sgx",
+		       &sgx_pdata),
+#endif
 #ifdef CONFIG_ARCH_OMAP4
 	OF_DEV_AUXDATA("ti,omap4-padconf", 0x4a100040, "4a100040.pinmux", &pcs_pdata),
 	OF_DEV_AUXDATA("ti,omap4-padconf", 0x4a31e040, "4a31e040.pinmux", &pcs_pdata),
diff --git a/include/linux/platform_data/sgx-omap.h b/include/linux/platform_data/sgx-omap.h
new file mode 100644
index 0000000..aa59b2c
--- /dev/null
+++ b/include/linux/platform_data/sgx-omap.h
@@ -0,0 +1,22 @@
+/*
+ * SGX Graphics Driver Platform Data
+ *
+ * Copyright (C) 2014 Texas Instruments Incorporated - http://www.ti.com/
+ *	Darren Etheridge <detheridge@ti.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed "as is" WITHOUT ANY WARRANTY of any
+ * kind, whether express or implied; without even the implied warranty
+ * of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+#include <linux/platform_device.h>
+
+struct gfx_sgx_platform_data {
+	const char *reset_name;
+
+	int (*deassert_reset)(struct platform_device *pdev, const char *name);
+};
-- 
2.5.0

